src/utils/hd-wallet.ts

```diff
@@ -3,9 +3,14 @@ import { full as hmac } from 'tweetnacl-auth'
 import { fromString } from 'bip32-path'
 import { decryptKey, encryptKey } from './crypto'
 import { encode } from './encoder'
-import {
-  InvalidDerivationPathError, NotHardenedSegmentError, UnsupportedChildIndexError
-} from './errors'
+import { CryptographyError } from './errors'
+
+export class DerivationError extends CryptographyError {
+  constructor (message: string) {
+    super(message)
+    this.name = 'DerivationError'
+  }
+}
```

Comment:
This is a new approach, I'm proposing to store errors related to specific functionality in the same file with it. This would allow to implement complex error classes without cycle dependencies. `DerivationError` would be still exported in the root of package because of #1058

Reply:
Got it. Types, constants, and errors will fall into the same file if referenced only in the same file.